app/routes.js

```diff
@@ -163,12 +163,32 @@ router.post('/ioj', function (req, res, next) {
     }
   } else {
     req.session.data['case_details_type'] = 'manual';
-    let offenceId = req.session.data['offence'];
-    if (offenceId) {
-      req.session.data['case_details']['offence'] = offencesList[offenceId].B;
-      req.session.data['case_details']['offence_class'] = offencesList[offenceId].D;
-      req.session.data['offence_search'] = offencesList[offenceId].B;
-    }
+    let offenceIds = req.session.data['offence'] || [];
+    let offences = []
+    let offenceSearches = []
+
+    offenceIds.forEach((offenceId, index) => {
+      let year = req.session.data['offence-year'][index]
+      let month = req.session.data['offence-month'][index]
+      let day = req.session.data['offence-day'][index]
+      if (offenceId) {
+        offences.push(
+          {
+            offence: offencesList[offenceId].B,
+            offence_class: offencesList[offenceId].D,
+            offence_date: `${year}-${month}-${day}`
```

Comment:
@oliverweighell I assumed this would be more helpful to have as an Obj for each offence and to match up the date of the offence as part of it too